packages/app/src/ui/organisms/ProgrammeProposalForm.tsx

```diff
@@ -22,6 +23,26 @@ import { Label as BaseLabel, LabelProps as BaseLabelProps } from "../Label";
 import { mdx } from "../mdx";
 import { Spacer } from "../Spacer";
 
+/**
+ * Type unsafe and impure 💩
+ *
+ * Don't use this function anywhere outside forms and use it only in places
+ * which are thoroughly tested.
+ *
+ * react-hook-form doesn't coerce values from number fields into numbers so we
+ * need to do this on submit. As we don't know the shape of the form value and
+ * it can be deeply nested, we use this dirty piece of code.
+ */
+function unsafeMod(obj: object, path: string, f: (x: unknown) => unknown) {
+  let cur = obj as any;
```

Comment:
What for this assertion is?

Reply:
This function is type unsafe. You can't index `object` with arbitrary keys.

You can find similar functions in _lodash_, and proper types for them look like this:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lodash/common/object.d.ts#L1659-L1792

See [monocle-ts](https://github.com/gcanti/monocle-ts) for typesafe alternative.